pkg/scheduler/objects/application.go

```diff
@@ -1276,12 +1276,21 @@ func (sa *Application) tryReservedAllocate(headRoom *resources.Resource, nodeIte
 	// lets try this on all other nodes
 	for _, reserve := range sa.reservations {
 		// Other nodes cannot be tried if the ask has a required node
-		if reserve.ask.GetRequiredNode() != "" {
+		ask := reserve.ask
+		if ask.GetRequiredNode() != "" {
 			continue
 		}
 		iterator := nodeIterator()
 		if iterator != nil {
-			alloc := sa.tryNodesNoReserve(reserve.ask, iterator, reserve.nodeID)
+			// check if this fits in the users' headroom first, if that fits check the queues' headroom
+			if !userHeadroom.FitInMaxUndef(ask.GetAllocatedResource()) {
+				continue
+			}
+			// check if this fits in the queue's head room
+			if !headRoom.FitInMaxUndef(ask.GetAllocatedResource()) {
+				continue
```

Comment:
Nit: simplify code:
```
if !headRoom.FinInMaxUndef() || !userHeadroom.FixInMaxUndef() {
   continue
}
```

Reply:
Thanks for the review @pbacsko I just had a new rev that addresses it. Would you please take a look again? thanks.